src/providers/utils.ts

```diff
@@ -1,8 +1,29 @@
 // The async/queue library has a task-based interface for building a concurrent queue.
-
+import assert from "assert";
 import { providers } from "ethers";
-import { isDefined } from "../utils";
 import { isEqual } from "lodash";
+import { isDefined } from "../utils";
+import * as alchemy from "./alchemy";
+import * as infura from "./infura";
+
+export type RPCProvider = "INFURA" | "ALCHEMY";
+
+const PROVIDERS = {
+  ALCHEMY: alchemy.getURL,
+  INFURA: infura.getURL,
+};
+
+export function isSupportedProvider(provider: string): provider is RPCProvider {
+  return ["ALCHEMY", "INFURA"].includes(provider);
+}
+
+export function getURL(provider: RPCProvider, chainId: number, apiKey?: string): string {
```

Comment:
Supplying `apiKey` should also be mandatory, and will avoid the need for an assert.

Reply:
Done: https://github.com/across-protocol/sdk/pull/704/commits/cfb1ccaaca4d9551831fabd6a92f98c66c58e77f